portfolio/src/main/webapp/script.js

```diff
@@ -293,9 +293,33 @@ function deleteSpecificComment(comment) {
 async function addComment() {
   const params = new URLSearchParams();
   const text = document.getElementById('text').value;
+  if (text === '') {
+    document.getElementById('text').value = '';
+    alert('Your message is empty. Please try again');
+    return;
+  }
   params.append('text', text);	
-  await fetch('/add-comment', {method: 'POST', body: params});
-  getCommentSection();
+  fetch(
+  'https://commentanalyzer.googleapis.com/v1alpha1/comments:analyze?key=API_KEY',
+    {
+      method: 'POST',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify({comment: {text: text}, languages: [], requestedAttributes: { TOXICITY: {} }})
+    }
+    ).then(async response => response.json()).then(async data => {
+      const score = data.attributeScores.TOXICITY.summaryScore.value;
+      if (score > 0.8) {
+        document.getElementById('text').value = '';
+        alert('Your message is too toxic. Please try again');
+        return;
+      }
+      else {
+        params.append('toxicity', score);
+        await fetch('/add-comment', {method: 'POST', body: params});
```

Comment:
We discussed this a bit earlier today, but are you able to change the body to a json string, using something like: `JSON.stringify({text: text, toxicity: toxicity})`?  This will also require a server side change to parse the data using gson, but should be more flexible for passing data other than strings, including arrays and objects

Reply:
Yes, I added this to the commit right after this one. I figured out why it seemed like the toxicity score was a `String`: the documentation of the `getParameter` method says it only returns a `String` or `null`. For floats like `toxicity`, it could have been fixed like this `Float toxicity = Float.parseFloat(request.getParameter("toxicity"));`, or using gson like i did below.